File: ancuta-gheorghe/RentingMovies/src/Rental.java
/**
 * Created with IntelliJ IDEA.
 * User: ancuta
 * Date: 5/18/13
 * Time: 5:41 PM
 * To change this template use File | Settings | File Templates.
 */
public class Rental {

    private Movie _movie;
    private int _daysRented;

    public Rental(Movie movie, int daysRented) {
        this._movie = movie;
        this._daysRented = daysRented;
    }

    public Movie getMovie() {
        return _movie;
    }

    public int getDaysRented() {
        return _daysRented;
    }

    public int getFrequentRenterPoints() {
        return _movie.getFrequentRenterPoints(_daysRented);
    }
}

File: cristina-condruz/RefactoringExamples/src/Rental.php
<?php
/**
 * Created by JetBrains PhpStorm.
 * User: Cristina Condruz
 * Date: 5/28/13
 * Time: 11:34 AM
 * To change this template use File | Settings | File Templates.
 */
require_once('Movie.php');
class Rental {
  private $_movie;//Movie type
  private $_daysRented;

  function __construct($_daysRented, Movie $_movie) {
    $this->_daysRented = $_daysRented;
    $this->_movie = $_movie;
  }

  public function getDaysRented() {
    return $this->_daysRented;
  }

  public function getMovie() {
    return $this->_movie;
  }

  /**
   * @return int
   */
  function getFrequentRenterPoints() {
     return $this->_movie->getFrequentRenterPoints($this->_daysRented);
  }

  /**
   * @return float|int
   */
  public function getCharge(){
       return $this->_movie->getCharge($this->_daysRented);
  }

}
